nixos/modules/services/network-filesystems/ipfs-cluster.nix

```diff
@@ -0,0 +1,123 @@
+    (optionalString (cfg.initPeers != [ ]) "--peers")
+    (lib.strings.concatStringsSep "," cfg.initPeers)
+  ];
+in {
+
+  ###### interface
+
+  options = {
+
+    services.ipfs-cluster = {
+
+      enable = mkEnableOption
+        "Pinset orchestration for IPFS - requires ipfs daemon to be useful";
+
+      user = mkOption {
+        type = types.str;
+        default = "ipfs";
+        description = "User under which the ipfs-cluster daemon runs";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "ipfs";
+        description = "Group under which the ipfs-cluster daemon runs";
+      };
+
+      consensus = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Consensus protocol - 'raft' or 'crdt'";
+      };
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/ipfs-cluster";
+        description = "The data dir for ipfs-cluster";
+      };
+
+      initPeers = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = "Peer addresses to initialize with on first run";
+      };
+
+      secret = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description =
+          "Secret for an existing cluster; if null, a new secret is generated";
+      };
+
+      secretFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description =
+          "File containing the secret - 'secret' and 'secretFile' should not both be set";
+      };
+
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.ipfs-cluster ];
+    environment.variables.IPFS_CLUSTER_PATH = cfg.dataDir;
+
+    systemd.tmpfiles.rules =
+      [ "d '${cfg.dataDir}' - ${cfg.user} ${cfg.group} - -" ];
+
+    systemd.packages = [ pkgs.ipfs-cluster pkgs.coreutils ];
+
+    systemd.services.ipfs-cluster-init = {
+      path = [ "/run/wrappers" pkgs.ipfs-cluster ];
+      environment = {
+        IPFS_CLUSTER_PATH = cfg.dataDir;
+        CLUSTER_SECRET = if (cfg.secret != null) then
+          cfg.secret elseif (secretFile != null) readFile cfg.secretFile
+        else
+          null;
+      };
+      wantedBy = [ "default.target" ];
+
+      serviceConfig = if cfg.consensus == null then
+        throw "ipfs-cluster requires the option 'consensus' to be set"
+      else {
+        ExecStart = [
+          ""
+          "${pkgs.ipfs-cluster}/bin/ipfs-cluster-service init --consensus ${cfg.consensus} ${initFlags}"
+        ];
+        Type = "oneshot";
+        RemainAfterExit = true;
+        User = cfg.user;
+        Group = cfg.group;
+      };
+      unitConfig.ConditionDirectoryNotEmpty = "!${cfg.dataDir}";
+    };
+
+    systemd.services.ipfs-cluster = {
+      path = [ "/run/wrappers" pkgs.ipfs-cluster ];
+      environment.IPFS_PATH = cfg.dataDir;
+      wantedBy = [ "default.target" ];
```

Comment:
Most services will be `wantedBy` `multi-user.target`. I'm not familiar with this daemon so can you please confirm?